internal/controllers/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -340,8 +341,8 @@ func TestMachineDeploymentReconciler(t *testing.T) {
 		// expect old MachineSets with old labels to be deleted
 		//
 		oldLabels := deployment.Spec.Selector.MatchLabels
-		oldLabels[clusterv1.MachineDeploymentLabelName] = deployment.Name
```

Comment:
oldLabels is only used to validate somewhere below that all MachineSets with the oldLabels have been deleted.

But those MachineSets will never have the `MachineDeploymentLabelName`set with the current code.

So this PR ensures we have a unique set of labels in selector which are then changed to other labels with no overlap.

This way we can clearly verify that the old MachineSets have been deleted. 

The test did not fail as:
* oldLabels never matched MachineSets vs. (current test)
* oldLabels don't match a MachineSet after MachineSet deletion (test after this PR)
are both accepted.


Please note: #7591 will ensure that `MachineDeploymentLabelName` will be set on all MachineSets, but this would break the current validation as the oldLabels would match old and new MachineSets.